contentgrid-automations-rest/src/main/java/com/contentgrid/automations/rest/AutomationModelVisitor.java

```diff
@@ -0,0 +1,149 @@
+                assertTwoTerms(terms);
+                yield Scalar.of(!terms.get(0).equals(terms.get(1)));
+            }
+            case AND -> {
+                assertNotEmpty(terms);
+                yield Scalar.of(terms.stream()
+                        .map(scalar -> scalar.assertResultType(Boolean.class))
+                        .map(ThunkExpression::maybeValue)
+                        .flatMap(Optional::stream)
+                        .map(Boolean.class::cast)
+                        .reduce(true, (left, right) -> left && right));
+            }
+            case OR -> {
+                assertNotEmpty(terms);
+                yield Scalar.of(terms.stream()
+                        .map(scalar -> scalar.assertResultType(Boolean.class))
+                        .map(ThunkExpression::maybeValue)
+                        .flatMap(Optional::stream)
+                        .map(Boolean.class::cast)
+                        .reduce(false, (left, right) -> left || right));
+            }
+            case NOT -> {
+                assertOneTerm(terms);
+                var term = (Scalar<Boolean>) terms.get(0).assertResultType(Boolean.class);
+                yield Scalar.of(!term.getValue());
+            }
+            default -> {
+                throw new UnsupportedOperationException("Operation %s not implemented"
+                        .formatted(functionExpression.getOperator().getKey()));
+            }
+        };
+    }
+
+    private static void assertNotEmpty(List<? extends ThunkExpression<?>> terms) {
+        if (terms.isEmpty()) {
+            throw new IllegalArgumentException("Operation requires at least 1 parameter.");
+        }
+    }
+
+    private static void assertOneTerm(List<? extends ThunkExpression<?>> terms) {
+        if (terms.size() != 1) {
+            throw new IllegalArgumentException("Operation requires 1 parameter.");
+        }
+    }
+
+    private static void assertTwoTerms(List<? extends ThunkExpression<?>> terms) {
+        if (terms.size() != 2) {
+            throw new IllegalArgumentException("Operation requires 2 parameters.");
+        }
+    }
+
+    @Override
+    public Scalar<?> visit(SymbolicReference symbolicReference, AutomationModel context) {
+        String subject = symbolicReference.getSubject().getName();
+        if (!SUBJECT_NAME.equalsIgnoreCase(subject)) {
+            throw new IllegalArgumentException(
+                    "Expected symbolic-ref subject named '" + SUBJECT_NAME + "', but got '" + subject + "'");
+        }
+
+        var path = symbolicReference.getPath();
+        try {
+            Object result = context;
+            for (var elem : path) {
+                Objects.requireNonNull(result, () -> "Cannot lookup property '%s' of null value"
+                        .formatted(getPathElementName(elem)));
+                var field = result.getClass().getDeclaredField(getPathElementName(elem));
+                field.setAccessible(true);
```

Comment:
Instead of handling reflection yourself directly, I would suggest taking a look at the Spring-supplied functionality, like [`PropertyAccessor`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/PropertyAccessor.html)